### DISCUSSION TITLE
Pointer.availablePhysicalBytes() Does not respect cgroup memory limit rule inside LXC container

### DISCUSSION THREAD

**Discussion by lastpeony** [bytedeco/javacpp#780] [Q&A]
I am running a java 17 application which is using org.bytedeco.javacpp.Pointer  v1.5.8 inside a LXC container.

I set cgroup 1 memory rule as around 2GB:
`lxc.cgroup.memory.limit_in_bytes = 214748364
`
Below calculation returns the correct memory, JVM respects the cgroup rule:
```
	public static long osTotalPhysicalMemory() {
		final OperatingSystemMXBean osBean = ManagementFactory.getOperatingSystemMXBean();
		if (osBean instanceof UnixOperatingSystemMXBean) {
			return (((UnixOperatingSystemMXBean) osBean).getTotalPhysicalMemorySize());
		} else {
			try {
				Method m = osBean.getClass().getDeclaredMethod("getTotalPhysicalMemorySize");
				m.setAccessible(true);
				return  (long) m.invoke(osBean);
			} catch (Exception e) {
				error(e);
				return -1L;
			}
		}
	}
```
For some reason `Pointer.availablePhysicalBytes()` returns 9889411072 -- 9.21 GB

Why? I think i am missing something. 

Thank you.

**Comment by saudet** [comment]
> Why? I think i am missing something.

Probably not: https://fabiokung.com/2014/03/13/memory-inside-linux-containers/

Contributions are welcome!
